### ISSUE TITLE
accuracy with ModelNet40

### ISSUE THREAD

**Issue by dogyoonlee** [HuguesTHOMAS/KPConv#108]
Hello

I really appreciate your great work.

However, I have questions related to **classification accuracy** on ModelNet40 with your default setting

All the things I revised are just the follows.

It's just the saving best accuracy, and best_vote accuracy process to compare the other works.

**_in trainer.py,_**

**1) validation part --> to save best accuracy**

revised self.validation like below and appended saving code

------------------------------------------
val_acc, vote_acc = self.validation(net, val_loader, config)

            if best_acc < val_acc:
                best_acc = val_acc
                conv_epoch = self.epoch
                if config.saving:
                    # Get current state dict
                    save_dict = {'epoch': self.epoch,
                                'model_state_dict': net.state_dict(),
                                'optimizer_state_dict': self.optimizer.state_dict(),
                                'saving_path': config.saving_path}

                    # Save current state of the network (for restoring purposes)
                    checkpoint_path = join(checkpoint_directory, 'best_acc_chkp.tar')
                    torch.save(save_dict, checkpoint_path)

            if best_vote_acc < vote_acc:
                best_vote_acc = vote_acc
                conv_epoch_vote = self.epoch
                if config.saving:
                    # Get current state dict
                    save_dict = {'epoch': self.epoch,
                                'model_state_dict': net.state_dict(),
                                'optimizer_state_dict': self.optimizer.state_dict(),
                                'saving_path': config.saving_path}

                    # Save current state of the network (for restoring purposes)
                    checkpoint_path = join(checkpoint_directory, 'best_vote_acc_chkp.tar')
                    torch.save(save_dict, checkpoint_path)
            
            print('>>> epoch: {:d} \nbest_acc: {:.3f} | conv_epoch: {:d} \nbest_vote_acc: {:.3f} | conv_epoch_vote: {:d} \n'.format(self.epoch, 
                                                                                best_acc, conv_epoch, 
                                                                                best_vote_acc, conv_epoch_vote))
                    
            with open(join(config.saving_path, 'best_results.txt'), "a") as file:
                message = '>>> epoch: {:d} \nbest_acc: {:.3f} | conv_epoch: {:d} \nbest_vote_acc: {:.3f} | conv_epoch_vote: {:d} \n'
                file.write(message.format(self.epoch,
                                            best_acc,
                                            conv_epoch,
                                            best_vote_acc,
                                            conv_epoch_vote))

------------------------------------------------------------

**2) revise validation function to return val_acc and vote_acc**
-------------------------------------------------
    def validation(self, net, val_loader, config: Config):

        if config.dataset_task == 'classification':
            _, val_acc, vote_acc = self.object_classification_validation(net, val_loader, config)
        elif config.dataset_task == 'segmentation':
            self.object_segmentation_validation(net, val_loader, config)
        elif config.dataset_task == 'cloud_segmentation':
            self.cloud_segmentation_validation(net, val_loader, config)
        elif config.dataset_task == 'slam_segmentation':
            self.slam_segmentation_validation(net, val_loader, config)
        else:
            raise ValueError('No validation method implemented for this network type')
        return val_acc, vote_acc
--------------------------------------------------------

**3) revise object_classification_validation function to return 3 values, C1, val_acc, vote_acc to use in validation function**
----------------------------------------------------------
                return C1, val_ACC, vote_ACC
------------------------------------------------------------

The code worked and experiment was successfully done.

The best validation accuracy was **93.75** and best vote_accuracy was **92.385** during the training

however, when I test the saved model of above training with test_models.py, 
the best test accuracy was **92.1** max

My questions is that which one is right results on your code?

as I check the data loading part of your code, 2 code use the same test_dataset which is used as val_dataset in training.
Also, the test process between 2 code is regarded almost same as I think.

what makes the accuracy different between 2 codes?
Did I experiment wrong?
